scipy/signal/firfilter.c

```diff
@@ -138,10 +138,17 @@ int pylab_convolve_2d (char  *in,        /* Input data Ns[0] x Ns[1] */
   if (outsize == FULL) {Os[0] = Ns[0]+Nwin[0]-1; Os[1] = Ns[1]+Nwin[1]-1;}
   else if (outsize == SAME) {Os[0] = Ns[0]; Os[1] = Ns[1];}
   else if (outsize == VALID) {Os[0] = Ns[0]-Nwin[0]+1; Os[1] = Ns[1]-Nwin[1]+1;}
-  else return -1;  /* Invalid output flag */  
+  else{
+    /* Invalid output flag */
+    free(indices);
```

Comment:
Instead of doing this dance twice, how about moving the allocation down below these checks.  There isn't a good reason to allocate the indices array if we end up not needing it.